Update FIPS documentation (#1536)
rancher/rke2: docs/security/fips_support.md

* crictl
* runc

## CNI

As of v1.21.2, RKE2 supports selecting a different CNI via the `--cni` flag and comes bundled with several CNIs including Canal (default), Calico, Cilium, and Multus. Of these, only Canal (the default) is rebuilt for FIPS compliance.

## Ingress

RKE2 ships with NGNIX as its default ingress provider. As of v1.21+, this component is FIPS compliant. There are two primary sub-components for NGINX ingress:

> - controller - responsible for monitoring Kubernetes and configuring NGINX accordingly

Reviewer: ```suggestion
- controller - responsible for monitoring/updating Kubernetes resources and configuring the server accordingly
```

Author: 👍

> - NGINX server- responsible for handling and routing traffic

Reviewer: ```suggestion
- server - responsible for accepting and routing traffic
```

Author: 👍

The controller is written in Go and as such is compiled using our [FIPS compatible Go compiler](./fips_support.md#use-of-fips-compatible-go-compiler).

> The NGINX server is written in C and also requires OpenSSL to function properly. As such, it leverages a FIPS-validated version of OpenSSL to achieve FIPS compliance.

Reviewer: ```suggestion
The server is written in C and also requires OpenSSL to function properly. As such, it leverages a FIPS-validated version of OpenSSL to achieve FIPS compliance.
```

Author: 👍
